tests/test_CompensatedGaussianFlux.py

```diff
@@ -0,0 +1,330 @@
+                        width,
+                        t,
+                    )
+
+                    # Compare values from c++ code to numpy code.
+                    self.assertFloatsAlmostEqual(flux, flux_numpy, rtol=1e-6)
+                    self.assertFloatsAlmostEqual(var, var_numpy, rtol=1e-6)
+
+                    # If the kernel width is equal to the simulated PSF then it
+                    # should be nearly unbiased.
+                    if width == psf_size:
+                        self.assertFloatsAlmostEqual(flux, true_flux, rtol=1e-3)
+
+                    # And check biases with non-zero DC offset; these should be
+                    # equal with some floating point tolerance.
+                    if dc_offset == 0.0:
+                        flux_0 = flux
+                        var_0 = var
+                    else:
+                        self.assertFloatsAlmostEqual(flux, flux_0, rtol=1e-7)
+                        self.assertFloatsAlmostEqual(var, var_0, rtol=1e-7)
+
+    def testCompensatedGaussianSubPixels(self):
+        """Test for correct instFlux as a function of sub-pixel position."""
+        np.random.seed(12345)
+
+        n_points = 100
+
+        x_sub = np.random.uniform(low=0.0, high=1.0, size=n_points)
+        y_sub = np.random.uniform(low=0.0, high=1.0, size=n_points)
+
+        # We need to set up the task in order to create the test dataset.
+        task = self.makeSingleFrameMeasurementTask("base_CompensatedGaussianFlux")
+
+        for width in (2, 3, 5):
+            for t in (1.5, 2.0):
+                fluxes = np.zeros(n_points)
+
+                for i in range(n_points):
+                    dataset = lsst.meas.base.tests.TestDataset(self.bbox_single)
+                    centroid = lsst.geom.Point2D(50.0 + x_sub[i], 50.0 + y_sub[i])
+                    dataset.addSource(50000.0, centroid)
+
+                    exposure, catalog = dataset.realize(10.0, task.schema, randomSeed=i)
+                    flux, var = _compensatedGaussianFiltInnerProduct(
+                        exposure.image.array,
+                        exposure.variance.array,
+                        centroid.getX(),
+                        centroid.getY(),
+                        width,
+                        t,
+                    )
+
+                    fluxes[i] = flux
+
+                # Check for no correlation with x_sub and y_sub.
+                fit_x = np.polyfit(x_sub, fluxes/50000.0, 1)
+                self.assertLess(np.abs(fit_x[0]), 1e-3)
+                fit_y = np.polyfit(y_sub, fluxes/50000.0, 1)
+                self.assertLess(np.abs(fit_y[0]), 1e-3)
+
+    def testCompensatedGaussianPlugin(self):
+        """Test for correct instFlux given known position and shape.
+        """
+        # In the z-band, HSC images have a noise of about 40.0 ADU, and a background
+        # offset of ~ -0.6 ADU/pixel.  This determines our test levels.
+        for width in (2, 3, 5):
```

Comment:
These numbers are occurring everywhere, would it be better to to the like somewhere in case they needed changed in tandem at some point in the future